radicle-node/src/service.rs

```diff
@@ -889,16 +895,61 @@ where
                     // which is required by the protocol to only announce refs it has.
                     if self.sessions.is_connected(announcer) {
                         match message.is_fresh(&self.storage) {
-                            Ok(is_fresh) => {
-                                if is_fresh {
-                                    // TODO: Only fetch if the refs announced are for peers we're tracking.
-                                    self.fetch(message.rid, announcer);
+                            Ok(true) => {
```

Comment:
Can we perhaps reduce the level of nesting in this code? It's pretty hard to read. I'd suggest eg. doing
```rust
if !self.sessions.is_connected(announcer) {
   debug!(...);
   return Ok(relay);
}
```
... and that sort of thing where it's easy.

Reply:
I've extracted the big inner blob into its a convenient method.